### PR TITLE
coverage subscriber write fields, missed name read field added

### PR DESCRIPTION
**Pull request by eclendenning-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java [hunk 1]
```diff
@@ -180,6 +183,36 @@ private boolean contactPointIsPhone(ContactPoint contactPoint) {
...
+    if (addresses.size() > 1) {
+      throw UnexpectedNumberOfValues.builder()
+          .exactExpectedCount(1)
+          .receivedCount(addresses.size())
+          .jsonPath(".contained[].address[]")
+          .build();
+    }
+    Address address = addresses.get(0);
+    Set<WriteableFilemanValue> addressValues = new HashSet<>();
+    factoryRegistry()
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java:199]: What about the rest of the subscriber address? Is there not already a method for this?

Comment (eclendenning-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java:199]: I reused the method where i could, but organization has less fields and more street lines. I then added these fields as a separate method as they are specific to the subscriber.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java [hunk 2]
```diff
@@ -629,12 +671,107 @@ WriteableFilemanValue phoneNumber(List<ContactPoint> telecom) {
...
+          .resourceType(RelatedPerson.class)
+          .id(referenceId)
+          .cause(e)
+          .build();
+    }
+  }
+
+  private WriteableFilemanValue relatedPersonBirthDate(String birthDate) {
+    if (isBlank(birthDate)) {
+      throw MissingRequiredField.builder().jsonPath(".contained[].birthDate()").build();
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java:733]: `()`?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java (hunk 1 above)
Comment (jhulbert-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageToInsuranceBufferTransformer.java:199]: I wonder if we should just overload the method and have this one call the other first then do the extra stuffs.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java [hunk 3]
```diff
@@ -394,15 +410,7 @@ private String subscriberBirthDate(FilemanEntry entry) {
...
-    ExtensionFactory extensions = ExtensionFactory.of(entry, YES_NO);
-    return emptyToNull(
-        Stream.of(
-                extensions.ofCodeableConceptFromInternalValue(
-                    InsuranceVerificationProcessor.INSUREDS_SEX,
-                    InsuranceBufferStructureDefinitions.INSUREDS_SEX_SYSTEM,
-                    InsuranceBufferStructureDefinitions.INSUREDS_SEX_URL))
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList()));
+    return Stream.of(birthsex(entry)).collect(Collectors.toList());
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/InsuranceBufferToR4CoverageTransformer.java:413]: I think we should just do `List.of()` for the time being.

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**